### ISSUE TITLE
Arrow navigation no longer works in dropdown with input

### ISSUE THREAD

**Issue by RobiNN1** [twbs/bootstrap#36598]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

After upgrade to the current 5.2.0 beta-1 arrow navigation (up/down) no longer works in dropdown with input. And it doesn't produce any errors in console so Im not sure where the problem is since it worked fine in v5.1.3.

### Reduced test cases

Open jsfiddle and click to input field, it opens dropdown menu with items but only in v5.1.3 you can navigate through items with up/down arrows.

BS 5.1.3 https://jsfiddle.net/2onqv40u/

BS 5.2.0 beta-1 https://jsfiddle.net/ot8nvqmL/

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

v5.2.0-beta-1

**Comment by jipexu** [comment]
according  to your 2 fiddles, same behavior here not work for (mac....  safari)

**Comment by SamuelMiller** [comment]
The issue is specific to the bootstrap's new 5.2 javascript plugin.
The dropdown works as expected when 5.2 beta css is paired with 5.1 javascript, as in the following:
```
<head>
<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor" crossorigin="anonymous">

<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>
</head>
```

**Comment by julien-deramond** [comment]
I'd say this issue comes from https://github.com/twbs/bootstrap/commit/bb7664db0aea2dddbc637992d2d0e78632dc2e68.

**Comment by GeoSot** [comment]
Guys, may I ask a question here?

Is there any link, on BS documentation, that provides an example with this markup as valid, or we are just discussing for a side effect that some devs took advantage of? 


**/CC** @patrickhlauke, do you think this is UX valid, to use this markup?

**Comment by RobiNN1** [comment]
It is not documented, but it should be. It's great feature for ajax search fields. Yes, it is possible to add own arrow navigation, but why not take advantage of that one in BS?

**Comment by patrickhlauke** [comment]
From memory, the old behavior is an old remnant/side effect from an early version of Bootstrap (possibly before version 4?) that was never properly documented.

The only official documentation regarding cursor keys appeared back in version 4  https://getbootstrap.com/docs/4.6/components/dropdowns/#accessibility

> However, Bootstrap does add built-in support for most standard keyboard menu interactions, such as the ability to move through individual `.dropdown-item` elements using the cursor keys and close the menu with the `ESC` key.

And I seem to remember the original code was quite hacky in terms of having to override normal keyboard behavior inside form fields that happen to be in a dropdown (e.g. things like textareas, where all of a sudden cursor keys didn't work as they should because of the override).

While I understand that having this undocumented feature helped create some interesting/creative approaches, like the ajax search above, note that those approaches are not fully "correct" from an accessibility point of view - you'd really want to implement them along these lines https://www.w3.org/WAI/ARIA/apg/patterns/combobox/

Even just for dropdowns with links, the current additional cursor navigation is "non-standard" really, but left there as a nice-to-have extra that doesn't interfere with regular accessibility of those dropdowns (which, in essence, are simply expand/collapse panels in the ARIA sense).

In short, I would suggest not reintroducing the old behavior and noting it as a change (but not a breaking change, since this was never an explicit/official feature of BS it seems).

**Comment by RobiNN1** [comment]
This was great undocumented feature but okay, I understand. I'll try to recreate arrow navigation with custom js.

**Comment by itsraghz** [comment]
> The issue is specific to the bootstrap's new 5.2 javascript plugin. The dropdown works as expected when 5.2 css beta is paired with 5.1 javascript, as in the following:
> 
> ```
> <head>
> <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor" crossorigin="anonymous">
> 
> <script defer src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>
> </head>
> ```

Perfect. This works as suggested. Hopefully  this would be permanently fixed in Bootstrap 5.2 soon.

**Comment by RobiNN1** [comment]
Ehm @itsraghz it will never be fixed, read properly comments above.
